### ISSUE TITLE
Edge case

### ISSUE THREAD

**Issue by samuelwilliams** [rlanvin/php-ip#48]
There is an interesting edge case whereby string floats between `"10.1"` and `"99.9"`  are interpreted as valid IP addresses by `inet_ntop` so the following does not throw an exception:
```php
$ip = new PhpIP\IPv4("55.5");
echo $ip->humanReadable(); //Echos "53.53.46.53"
```
This is demonstrated as a test here [https://travis-ci.org/samuelwilliams/php-ip/builds/510198479](https://travis-ci.org/samuelwilliams/php-ip/builds/510198479)

**Comment by samuelwilliams** [comment]
Also demonstrated in this code:

```php
for ($i=0;$i<=1100;$i++) {
    $n = sprintf('%.1f', 0.1 * $i);
    echo $n .'~ ' . @inet_ntop((string) $n) ?: '';
    echo PHP_EOL;
}
```

**Comment by rlanvin** [comment]
Hey @samuelwilliams I see you closed this issue. Is it because you think it's not worth investigating or fixing? It sounds like a strange behavior of PHP, so indeed I'm not sure what to do with it.

**Comment by samuelwilliams** [comment]
Some packed IP addresses happen to be the same bit arrangement as some floats, case in point:
```php
echo inet_pton('53.53.46.53'); //Echos "55.5"
echo unpack('H*', 53.3)[1]; //Echos "35332e33" ('53.53.46.53' in hex)
```
When I came back to the issue to re-examine it, I found it to not be a big deal (I don't think it actually constitutes a bug). The only time it would actually cause issues is if there was some kind of user input that has not been validated. The fix is simple, validate the input.
